jiant/models.py

```diff
@@ -580,23 +587,26 @@ def build_image_sent_module(task, d_inp, params):
     return pooler
 
 
-def build_single_sentence_module(task, d_inp: int, use_bert: bool, params: Params):
+def build_single_sentence_module(task, d_inp: int, use_pytorch_transformers: bool, params: Params):
     """ Build a single sentence classifier
 
     args:
         - task (Task): task object, used to get the number of output classes
         - d_inp (int): input dimension to the module, needed for optional linear projection
-        - use_bert (bool): if using BERT, skip projection before pooling.
+        - use_pytorch_transformers (bool): if using BERT/XLNet, skip projection before pooling.
```

Comment:
This seems like an overly general name for a parameter with a narrow function - could we rename as 'project_before_pooling' and just set this to false in the calling code, so that this function doesn't need to be explicitly aware of the encoder type?

Reply:
I don't think we enforce this anywhere else, but the Goldberg setup is also fairly unusual and delicate experimental, so I don't want to give the impression that it's properly supported until we've tested it. I revised the comment.

Reply:
(Push coming soon)

Reply:
I agree with Ian, though don't feel too strongly about it.

Reply:
The logic doesn't seem quite that simple. E.g., line 666. It couldn't hurt to clean it up, but I'm inclined to leave it—this works, and I don't know all of the logic here well enough to know what to watch out for when cleaning.

Reply:
I also have some concerns on this, use_pytorch_transformer includes not only BERT/XLNet, but also GPT(2), TransformerXL. It is not using pytorch transformer that causes the difference in output module (and process split), but using models that support integrated ( two sentences) embedding.
We probably need to differentiate between logic those apply when using two sentence embedding and those apply when using pytorch transformers, otherwise there will be a lot of confusion later on.

Reply:
Cleaned up a bit. We'll definitely need to make this more complex when we add GPT/TXL support, but we're not there yet.